src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -785,9 +785,15 @@ String getHostName(boolean check) {
     }
 
     /**
-     * Gets the fully qualified domain name for this IP address.
-     * Best effort method, meaning we may not be able to return
-     * the FQDN depending on the underlying system configuration.
+     * Gets the fully qualified domain name for this
+     * {@linkplain InetAddress#getAddress() IP address} using the system-wide
+     * {@linkplain InetAddressResolver resolver}.
+     *
+     * <p>The system-wide resolver will be used to do a reverse name lookup of the IP address.
+     * The lookup may fail for reasons such as the host not being registered with the name
+     * service. In such cases, where the resolver isn't able to determine the fully qualified
+     * domain name, this method returns the {@linkplain #getHostAddress() textual representation}
+     * of the IP address.
```

Comment:
I think it would be better to say that the lookup can fail for many reasons that include the host not being registered with the name service. 

A suggestion for "In such cases .." is to say "f the resolve is unable to determine ...". I think that would be a bit easer to read.

Reply:
That sounds good. I've updated the PR to use this suggestion.